weblate/trans/views/edit.py

```diff
@@ -117,6 +117,8 @@ def get_other_units(unit):
         return result
 
     for item in units:
+        if item.source == "":
+            continue
```

Comment:
Better to do this in the database query above to avoid fetching not needed units from the database. Appending `.exclude(source="")` should do this.

Reply:
yep, tested, works, fixed, pushed, thanks.

I'd like to help with a unittest but I absolutely have no expierience in python.